MainRepository/Program.cs

```diff
@@ -1,7 +1,54 @@
-﻿internal class Program
-{
-    private static void Main(string[] args)
-    {
-        Console.WriteLine("Hello, World!");
-    }
-}
\ No newline at end of file
+﻿// This program validates the file path
+// BNF diagram of finding valid file path
+using static State;
+var strs = new List<string> { @"C:\PROGRAMS\DATA\README.TXT", @"C:\PROGRAMS\README.TXT"
+   , @"C:\PROGRAMS\DATA\DIR\README.TXT", @"C:\README.TXT", "..", "", "C:"
+   , "C:\\PROGRAMS\\README..TXT" };
+foreach (var str in strs) {
+   if (IsValidFilePath (str)) {
+      PrintPath (str, ConsoleColor.Green);
+      var (dir, folder, filename, ext) = Result (str);
+      Console.WriteLine ($"{dir}, {folder}, {filename}, {ext}");
+   } else {

```

Comment:
```suggestion
   } else PrintPath (str, ConsoleColor.Red);

```
@karthick-1205 Single line body of else or if doesn't need parentheses